R/gh_fill.R

```diff
@@ -0,0 +1,28 @@
+#' Fill geohash prefix with members
+#'
+#' @param geohashes Character vector of input geohashes. They must all be of same precision
+#' @param precision Positive integer scalar controlling the 'zoom level' – how many characters should be used in the output.
+#' @return Character vector of geohashes corresponding to the input.
+#' @export
+
+
+
+gh_fill <- function(geohashes, precision) {
+  if (uniqueN(nchar(geohashes)) > 1) {
+    stop("Input Geohashes must all have the same precision level.")
+  }
+  if (sum(grepl("['ailo]", geohashes)) > 0) {
+    stop("Invalid Geohash; Valid characters: [0123456789bcdefghjkmnpqrstuvwxyz]")
+  }
+
+  new_levels <- precision - nchar(geohashes[1])
+
+  base32 <-
+    unlist(strsplit("0123456789bcdefghjkmnpqrstuvwxyz", split = ""))
+
+  grid <-
+    do.call(data.table::CJ, append(list(geohashes), replicate(new_levels, base32, FALSE)))
```

Comment:
could you do a performance comparison of `CJ()` vs. `expand.grid()` for typical inputs? E.g. 1e4 inputs and adding 1-3 characters of precision...

Depending on that we'll add the data.table dependency